src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/ConstantExpectedAnalyzer.UnmanagedHelper.cs

```diff
@@ -0,0 +1,350 @@
+                    return false;
+                }
+                integer = Convert.ToInt64(constant);
+            }
+            catch
+            {
+                integer = default;
+                return false;
+            }
+            return true;
+        }
+        private static bool TryConvertUnsignedInteger(object constant, out ulong integer)
+        {
+            try
+            {
+                if (constant is string or bool)
+                {
+                    integer = default;
+                    return false;
+                }
+                integer = Convert.ToUInt64(constant);
+            }
+            catch
+            {
+                integer = default;
+                return false;
+            }
+            return true;
+        }
+
+        private static bool TryTransformInt64(object constant, out long value, out bool isInvalid)
+        {
+            bool isValidSigned = TryConvertSignedInteger(constant, out value);
+            isInvalid = false;
+            if (isValidSigned)
+            {
+                return isValidSigned;
+            }
+            if (!TryConvertUnsignedInteger(constant, out _))
+            {
+                isInvalid = true;
+            }
+            return isValidSigned;
+        }
+        private static bool TryTransformUInt64(object constant, out ulong value, out bool isInvalid)
+        {
+            bool isValidUnsigned = TryConvertUnsignedInteger(constant, out value);
+            isInvalid = false;
+            if (isValidUnsigned)
+            {
+                return isValidUnsigned;
+            }
+            if (!TryConvertSignedInteger(constant, out _))
+            {
+                isInvalid = true;
+            }
+            return isValidUnsigned;
+        }
+
+        private static bool TryTransformChar(object constant, out char value, out bool isInvalid)
+        {
+            try
+            {
+                if (constant is string or bool)
+                {
+                    return Invalid(out value, out isInvalid);
+                }
+                value = Convert.ToChar(constant);
+            }
+            catch
+            {
+                return Invalid(out value, out isInvalid);
+            }
+            isInvalid = false;
+            return true;
+        }
+
+        private static bool TryTransformBoolean(object constant, out bool value, out bool isInvalid)
+        {
+            if (constant is bool b)
+            {
+                value = b;
+                isInvalid = false;
+                return true;
+            }
+            return Invalid(out value, out isInvalid);
+        }
+
+        private static bool TryTransformSingle(object constant, out float value, out bool isInvalid)
+        {
+            if (constant is string or bool)
+            {
+                return Invalid(out value, out isInvalid);
+            }
+            value = Convert.ToSingle(constant);
```

Comment:
Is constant nullable?
If yes, then that is fine.
If no, I think float.Parse would be more appropriate.